### PR TITLE
All files table

### PR DESCRIPTION
**Pull request by binayak20**
Files table issue fixed

### CODE REVIEW

File: app/project/[id]/components/project-task-files.tsx [hunk 1]
```diff
@@ -25,14 +29,43 @@ const filesColumns: Tan.ColumnDef<DesignFile>[] = [
...
-    cell: ({ row }) => <pre>{row.getValue("size")}</pre>,
+    cell: ({ row }) => <pre>{getFileSize(row.getValue("filesize"))}</pre>,
   },
   {
     accessorKey: "taskid",
     header: () => <div>Task Id</div>,
-    cell: ({ row }) => <pre>{row.getValue("taskid")}</pre>,
+    cell: ({ row, projectid }: any) => (
+      <Button asChild variant="outline" className="h-8 w-8 p-0">
+        <Link href={`/project/${projectid}/task/${row.getValue("taskid")}`}>
```
Comment (patrickchin) [inline app/project/[id]/components/project-task-files.tsx:64]: confusingly this should actually be specid
so each taskid is unique but each project can have similar specids e.g. /project/xxx/task/100 will always be "Title Search" etc..

Comment (patrickchin) [reply app/project/[id]/components/project-task-files.tsx:64]: Also if there is a comment linked with this file, i'd like to link to that comment e.g.
https://haru2-kappa.vercel.app/project/365/task/100#comment-137

File: app/project/[id]/components/project-task-files.tsx [hunk 2]
```diff
@@ -44,33 +77,41 @@ const filesColumns: Tan.ColumnDef<DesignFile>[] = [
...
     getPaginationRowModel: Tan.getPaginationRowModel(),
+    getSortedRowModel: Tan.getSortedRowModel(),
+    getFilteredRowModel: Tan.getFilteredRowModel(),
+    state: {
+      sorting,
+      columnFilters,
+    },
   });
+  console.log("Data:", data);
+  console.log("projectid:", projectid);
```
Comment (patrickchin) [inline app/project/[id]/components/project-task-files.tsx:110]: these can be removed :)

File: lib/utils.ts [hunk 3]
```diff
@@ -18,3 +18,10 @@ export function getAvatarInitials(fullname?: string): string {
 
   return initials.toUpperCase();
 }
+
+export function getFileSize(size?: number): string {
+  if (!size) return "";
+  if (size < 1000) return `${size} KB`;
```
Comment (patrickchin) [inline lib/utils.ts:24]: This looks incorrect, maybe we can use https://www.npmjs.com/package/pretty-bytes instead :D

File: app/settings/page.tsx [hunk 4]
```diff
@@ -60,6 +60,10 @@ function SettingsPage() {
     location.reload();
   };
 
+  const deleteProfileAvatar = () => {
+    //TODO
```
Comment (patrickchin) [inline app/settings/page.tsx:64]: This is still in TODO!

Comment (binayak20) [reply app/settings/page.tsx]: Pushed all the fixations and complete the task

Comment (binayak20) [reply app/settings/page.tsx]: If any feedback please let me know

File: components/delete-confirmation.tsx
Comment (patrickchin) [inline components/delete-confirmation.tsx:1]: nice addition! but I'd prefer if we keep using the shadcn library for these things:
https://ui.shadcn.com/docs/components/alert-dialog

Comment (binayak20) [reply components/delete-confirmation.tsx:1]: Changed

File: drizzle/schema.ts [hunk 5]
```diff
@@ -121,6 +121,7 @@ export const files1 = pgTable("files1", {
   uploaderid: integer("uploaderId").references(() => users1.id),
   projectid: integer("projectId").references(() => projects1.id),
   taskid: integer("taskId").references(() => tasks1.id),
+  specid: integer("specid"),
```
Comment (patrickchin) [inline drizzle/schema.ts:124]: I'd prefer not to have this here as there is now, no one source of truth. To get the specid I guess you'd have to join the files1 and the tasks1 tables and  the specid would be accessible under `file.task.specId`

Comment (binayak20) [reply drizzle/schema.ts]: change the specId source of truth

**Comment by patrickchin** [comment]
finally addressed all the silly typescript errors I introduced

**Pull request merged**